FrontEnd/source/board.js

```diff
@@ -0,0 +1,170 @@
+  state.gameBoardRows = newStateValues.gameBoardRows;
+}
+
+function resetBoard(row, column) {
+  $('#game-board').html('');
+
+  const body = {
+    gameBoardValues: state.gameBoardValues,
+    currentPlayer: 'red',
+    gameBoardColumns: column,
+    gameBoardRows: row,
+  };
+  $.ajax({
+    type: 'POST',
+    url: '/resetBoard',
+    data: JSON.stringify(body),
+    contentType: 'application/json',
+    success: (result) => {
+      const resultObject = result.updatedState;
+      updateStateValues(resultObject);
+
+      $('#player-counter').css('background-color', state.currentPlayer);
+    },
+  });
+}
+
+function removeHover() {
+  for (let i = 0; i < state.gameBoardColumns; i += 1) {
+    $(`#${i}`).removeClass('column-hover');
+  }
+}
+
+function disableBoard() {
+  for (let i = 0; i < state.gameBoardColumns; i += 1) {
+    $(`#${i}`).off('click');
+  }
+}
+
+function showWinner(winningColor) {
+  $('.fireworks').attr('style', 'display: block;');
+  $('.winner-text').text(`${winningColor.toUpperCase()} TEAM WINS!`);
+  $('.winner-text').css('color', winningColor);
+  const winCount = $(`#${winningColor}-win-count`).text();
+  $(`#${winningColor}-win-count`).text(parseInt(winCount, 10) + 1);
+  disableBoard();
+  removeHover();
+}
+
+function checkWin(columnRowIndex) {
+  const columnRowArray = columnRowIndex.split('-');
+  const body = {
+    column: columnRowArray[0],
+    row: columnRowArray[1],
+  };
+
+  $.ajax({
+    type: 'POST',
+    url: '/checkWin',
+    data: JSON.stringify(body),
+    contentType: 'application/json',
+    success: (result) => {
+      if (result.winDetected != null) {
+        showWinner(result.winDetected);
+      }
+    },
+  });
+}
+
+function placeCounter(columnIndex) {
+  let newlyPlacedPiece;
+  const body = {
+    column: columnIndex,
+  };
+  $.ajax({
+    type: 'POST',
+    url: '/placeCounter',
+    data: JSON.stringify(body),
+    contentType: 'application/json',
+    success: (result) => {
+      const resultObject = result.updatedState;
+      newlyPlacedPiece = result.placedCounterPos;
+
+      $(`#${newlyPlacedPiece}`).css('background-color', state.currentPlayer);
+
+      updateStateValues(resultObject);
+
+      $('#player-counter').css('background-color', state.currentPlayer);
+
+      checkWin(newlyPlacedPiece);
+    },
+  });
+}
+
+function drawBoardToHtml(columnSize, rowSize) {
+  for (let i = 0; i < columnSize; i += 1) {
+    const divColumn = $('<div></div>');
+    divColumn.click(() => {
+      placeCounter(i);
+    });
+    divColumn.attr('class', 'column column-hover flex-container');
+    divColumn.attr('id', i);
+
+    for (let j = 0; j < rowSize; j += 1) {
+      const divRow = $('<div></div>');
+      divRow.attr('class', 'row');
+      divRow.attr('id', `${i}-${j}`);
+      divColumn.append(divRow);
+    }
+
+    $('#game-board').append(divColumn);
+  }
+}
+
+function displayInGameAssets() {
+  $('#win-count-area').attr('style', 'display: block');
```

Comment:
You might like to know that jQuery has `.show()` and `.hide()` to make this easier